### DISCUSSION TITLE
Hi all! Does a Rule exists for this Rename StaticClass::a();   to StaticClass::b()->someMethod(); ?

### DISCUSSION THREAD

**Discussion by alexandreelise** [rectorphp/rector#7342] [Q&A]
Hi all! Does a Rule exists for this Rename StaticClass::a();   to StaticClass::b()->someMethod(); ?

Or should I create this Rule with rector-generator?

**Comment by TomasVotruba** [comment]
Hi,

we don't have such rule yet, so you'll have to make it yourself.
I'm curious, what do you need this for? :)

**Reply by alexandreelise** [reply]
It's to migrate Joomla from 3.10.10 to 4.1.5. I'm learning Rector seriously since it's an awesome tool and saves a lot of time. We discussed some time ago about Joomla migration from 3 to 4 and you told me to do it myself as for now It's not useful to as many people as symfony or laravel and other leading frameworks. So I have to get my hands dirty drink some orange juice or cold water and make my neurons work. Thanks for this awesome tool.

**Reply by TomasVotruba** [reply]
That sounds cool :sunglasses: 

> We discussed some time ago about Joomla migration from 3 to 4 and you told me to do it myself as for now It's not useful to as many

I'd probably to rephrase that now: if you understand Joomla, you can do it yourself and publish as an open-source package to help the whole community :+1:  Similar work has been done in many communities (good for inspiration on "how to" start):

* https://github.com/palantirnet/drupal-rector
* https://github.com/craftcms/rector
* https://github.com/sulu/sulu-rector
* https://github.com/FriendsOfShopware/shopware-rector
* https://github.com/sabbelasichon/typo3-rector

> So I have to get my hands dirty drink some orange juice or cold water and make my neurons work. Thanks for this awesome tool.

Go for it :+1: The start is slow, but in a month or two, you'll be able to turn any BC change to a Rector rule in an hour.

**Reply by alexandreelise** [reply]
Hi @TomasVotruba If you would have to migrate our project from Joomla 3.10.10 to Joomla 4.1.5 how much would you charge? We need it to be done in 2 weeks maximum

**Reply by alexandreelise** [reply]
Someone in the joomla community already dit what you suggested https://github.com/nikosdion/joomlatypehints

**Reply by TomasVotruba** [reply]
> Hi @TomasVotruba If you would have to migrate our project from Joomla 3.10.10 to Joomla 4.1.5 how much would you charge? We need it to be done in 2 weeks maximum

I'll reply in PM
